core/glrender/src/main/java/edu/cmu/cs/dennisc/render/gl/imp/adapters/GlrSymmetricPerspectiveCamera.java

```diff
@@ -63,27 +63,23 @@
 public class GlrSymmetricPerspectiveCamera extends GlrAbstractPerspectiveCamera<SymmetricPerspectiveCamera> {
   @Override
   public Ray getRayAtPixel(Ray rv, int xPixel, int yPixel, Rectangle actualViewport) {
-    double vertical = getActualVerticalViewingAngle(actualViewport).getAsRadians();
-    double near = -owner.nearClippingPlaneDistance.getValue();
-    double far = -owner.farClippingPlaneDistance.getValue();
+    final double near = owner.nearClippingPlaneDistance.getValue();
+    final double far = owner.farClippingPlaneDistance.getValue();
 
-    //todo: investigate (especially x)
-    xPixel = actualViewport.width - xPixel + actualViewport.x;
-    yPixel = actualViewport.height - yPixel - actualViewport.y;
-
-    final double tanHalfVertical = Math.tan(vertical * 0.5);
-    final double aspect = getAspectRatio(actualViewport);
-    final double halfWidth = actualViewport.width * 0.5;
-    final double halfHeight = actualViewport.height * 0.5;
-    final double dx = tanHalfVertical * ((xPixel  / halfWidth) - 1.0) * aspect;
-    final double dy = tanHalfVertical * (1.0 - (yPixel / halfHeight));
+    // xPixel and yPixel are given to us with 0, 0 in the upper left of the viewport, so flip y.
+    // actualViewport.x & y are set > 0 when letterboxing
+    final double xRatio = (xPixel - actualViewport.x) / ( actualViewport.width * 0.5);
+    final double yRatio = (actualViewport.height - yPixel - actualViewport.y) / ( actualViewport.height * 0.5);
+    final double tanHalfVertical = Math.tan(getActualVerticalViewingAngle(actualViewport).getAsRadians() * 0.5);
+    final double dx = (1.0 - xRatio) * tanHalfVertical * getAspectRatio(actualViewport);
+    final double dy = (1.0 - yRatio) * tanHalfVertical;
```

Comment:
You have made it so simple a five year old can understand it.
(Go get me a five year old, I can't make heads or tails of it)

Reply:
Please pardon my smart ass comment.
It took a little, but I think I follow before and after - and I prefer after.
Also, I used it and it works.